src/view/renderer.js

```diff
@@ -243,6 +236,67 @@ export default class Renderer {
 		this.markedChildren.clear();
 	}
 
+	/**
+	 * Updates viewElement children mappings. Children which were replaced in the view structure by the similar
+	 * element (same tag name) are treated as 'replaced'. Their mappings are rebind to the corresponding,
+	 * existing DOM element so they will not be replaced by a new DOM element during rendering.
+	 *
+	 * @private
+	 * @param {module:engine/view/node~Node} viewElement The view element which children mappings will be updated.
+	 */
+	_updateChildrenMappings( viewElement ) {
+		// We do not perform any operations on DOM here so there is no need to bind view element or convert its children.
+		const diff = this._diffElementChildren( viewElement, { bind: false, withChildren: false } );
+
+		if ( diff ) {
+			const actions = this._findReplaceActions( diff.actions, diff.actualDomChildren, diff.expectedDomChildren );
+
+			if ( actions.indexOf( 'replace' ) !== -1 ) {
+				const counter = { equal: 0, insert: 0, delete: 0 };
+				for ( const action of actions ) {
```

Comment:
There should be a blank line before `for()`/`if()`/`while`/`function`. Odd that we don't have this covered in ESLint.

Reply:
I suppose I wasn't fully aware of that, but I will keep this in mind now 👍